modules/afsmtp/afsmtp.c

```diff
@@ -119,51 +116,56 @@ afsmtp_dd_set_port(LogDriver *d, gint port)
 }
 
 void
-afsmtp_dd_set_subject(LogDriver *d, const gchar *subject)
+afsmtp_dd_set_subject(LogDriver *d, const gchar *subject, GlobalConfig *cfg)
 {
   AFSMTPDriver *self = (AFSMTPDriver *)d;
 
-  g_free(self->subject);
-  self->subject = g_strdup(subject);
+  log_template_unref(self->subject_tmpl);
+  self->subject_tmpl = log_template_new(cfg, "subject");
+  self->subject_tmpl->template = g_strdup(subject);
 }
```

Comment:
I poked into that member, because you suggested that we should get rid of the string representations of the templated strings. 
In my latest commit bc1a25203612c60faefe5fe61265908f51e62568 I used the template_content rule and it takes care of calling log_template_new, log_template_compile and also takes care of error handling, so if the log_template_compile function runs to an invalid template, syslog-ng fails the startup.